### PR TITLE
Kadane's Algorithm

### PR DESCRIPTION
**Pull request by tk315**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Kadane algorithm article added.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#207

### CODE REVIEW

**Comment by tk315** [comment]
Please review @HarshCasper @ankitaggarwal23 @iamrajiv @atarax665

**Review by iamrajiv — APPROVED** [review]

File: docs/dynamic-programming/kadane-algorithm.mdx [hunk 1]
```diff
@@ -0,0 +1,222 @@
+---
+title: Kadane's Algorithm
+slug: /dp/kadane-algorithm
```
Comment (HarshCasper) [inline docs/dynamic-programming/kadane-algorithm.mdx:3]: It should be `dynamic-programming`

File: docs/dynamic-programming/kadane-algorithm.mdx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+-5
+```
+#### Output :
+```
+Maximum sum of contiguous subarray: 7
+```
+
+## Complexity Analysis
+
+`n` is the size of array.
```
Comment (HarshCasper) [inline docs/dynamic-programming/kadane-algorithm.mdx:209]: Keep it as a bullet point

File: docs/dynamic-programming/kadane-algorithm.mdx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+#### Input :
+
+```
+Enter the size of array: 5
+Enter the elements of array: 
+1 
+-2 
+3 
+4 
+-5
```
Comment (HarshCasper) [inline docs/dynamic-programming/kadane-algorithm.mdx:200]: Left-align all of them

File: docs/dynamic-programming/kadane-algorithm.mdx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+## Algorithm
+
+- Initialize `max_sum` and `sum_till_now` to `arr[0]`.
+
+- Initialize `i to 1` begin a While loop till `i != n`
+  - If `(sum_till_now < 0)` set it to `0`
+  - `sum_till_now += arr[i]`
+  - `max_sum` = `maximum(max_sum, sum_till_now)`
+  - `i += 1`.
+- Return `max_sum`.
```
Comment (HarshCasper) [inline docs/dynamic-programming/kadane-algorithm.mdx:72]: Use text for writing the algorithm. You can add a separate pseudo-code section but it isn't necessary.

File: docs/dynamic-programming/kadane-algorithm.mdx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+Kadane's Algorithm gives us the maximum sum out of all possible contiguous subarray in a one-dimensional array. It is a Dynamic Programming based algorithm because it uses the previously computed data for further computations. It is the most optimal solution for finding the maximum sum contiguous subarray. It has many applications like genomic sequence analysis and computer vision.
+
+## Explanation 
+
+Contiguous subarray refers to the continuous part of an array. Contiguous subarrays of `{p,q,r}` are `{p}`, `{q}`, `{r}`, `{p,q}`, `{q,r}`, `{p,q,r}`. This algorithm iterates over all positive-sum contiguous subarrays and keep a track of the maximum sum contiguous subarray. If the sum of the current subarray on which we are iterating is greater than the maximum sum, then we will replace the maximum sum with the current sum.
+
+Let's have one example: 
+<img src="https://i.imgur.com/DKoovMD.png" width="365" height="63" align="center"/><br/><br/>
+
+```
```
Comment (HarshCasper) [inline docs/dynamic-programming/kadane-algorithm.mdx:17]: Write proper descriptions for the variables you are having.

File: docs/dynamic-programming/kadane-algorithm.mdx [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+## Introduction
+
+Kadane's Algorithm gives us the maximum sum out of all possible contiguous subarray in a one-dimensional array. It is a Dynamic Programming based algorithm because it uses the previously computed data for further computations. It is the most optimal solution for finding the maximum sum contiguous subarray. It has many applications like genomic sequence analysis and computer vision.
+
+## Explanation 
+
+Contiguous subarray refers to the continuous part of an array. Contiguous subarrays of `{p,q,r}` are `{p}`, `{q}`, `{r}`, `{p,q}`, `{q,r}`, `{p,q,r}`. This algorithm iterates over all positive-sum contiguous subarrays and keep a track of the maximum sum contiguous subarray. If the sum of the current subarray on which we are iterating is greater than the maximum sum, then we will replace the maximum sum with the current sum.
+
+Let's have one example: 
+<img src="https://i.imgur.com/DKoovMD.png" width="365" height="63" align="center"/><br/><br/>
```
Comment (HarshCasper) [inline docs/dynamic-programming/kadane-algorithm.mdx:15]: Did you make the image?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Also remove redundant comments

File: docs/dynamic-programming/kadane-algorithm.mdx (hunk 6 above)
Comment (tk315) [reply docs/dynamic-programming/kadane-algorithm.mdx:15]: yes

File: docs/dynamic-programming/kadane-algorithm.mdx (hunk 1 above)
Comment (tk315) [reply docs/dynamic-programming/kadane-algorithm.mdx]: done

File: docs/dynamic-programming/kadane-algorithm.mdx (hunk 2 above)
Comment (tk315) [reply docs/dynamic-programming/kadane-algorithm.mdx]: done

File: docs/dynamic-programming/kadane-algorithm.mdx (hunk 3 above)
Comment (tk315) [reply docs/dynamic-programming/kadane-algorithm.mdx]: done

File: docs/dynamic-programming/kadane-algorithm.mdx (hunk 4 above)
Comment (tk315) [reply docs/dynamic-programming/kadane-algorithm.mdx]: done

File: docs/dynamic-programming/kadane-algorithm.mdx (hunk 5 above)
Comment (tk315) [reply docs/dynamic-programming/kadane-algorithm.mdx:17]: done

**Review by atarax665 — APPROVED** [review]

**Comment by atarax665** [comment]
Resolve conflict.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**